Wox/ViewModel/SettingWindowViewModel.cs

```diff
@@ -47,6 +47,42 @@ public async void UpdateApp()
             await _updater.UpdateApp(false);
         }
 
+        // This is only required to set at startup. When portable mode enabled/disabled a restart is always required
+        private bool _portableMode = DataLocation.PortableDataLocationInUse();
+        public bool PortableMode
+        {
+            get { return _portableMode; }
+            set
+            {
+                if (!_portable.CanUpdatePortability())
```

Comment:
can we somehow move all the logic to the portable class? so one place will handle the logic, or maybe somehow make this a command as it does many things now?